src/squirrel/supertux_api.cpp

```diff
@@ -0,0 +1,794 @@
+
+namespace Globals {
+
+/**
+ * Displays the value of an argument. This is useful for inspecting tables.
+ * @param ANY $object
+ */
+static void display(const ssq::Object& object)
+{
+  ConsoleBuffer::output << squirrel_to_string(object) << std::endl;
+}
+/**
+ * Displays the contents of the current stack.
+ */
+static void print_stacktrace(HSQUIRRELVM vm)
+{
+  print_squirrel_stack(vm);
+}
+/**
+ * Returns the currently running thread.
+ */
+static SQInteger get_current_thread(HSQUIRRELVM vm)
+{
+  sq_pushthread(vm, vm);
+  return 1;
+}
+
+/**
+ * Returns whether the game is in christmas mode.
+ */
+static bool is_christmas()
+{
+  return g_config->christmas_mode;
+}
+
+/**
+ * Starts a skippable cutscene.
+ */
+static void start_cutscene()
+{
+  auto session = GameSession::current();
+  if (session == nullptr)
+  {
+    log_info << "No game session" << std::endl;
+    return;
+  }
+
+  if (session->get_current_level().m_is_in_cutscene)
+  {
+    log_warning << "start_cutscene(): starting a new cutscene above another one, ending preceding cutscene (use end_cutscene() in scripts!)" << std::endl;
+
+    // Remove all sounds that started playing while skipping.
+    if (session->get_current_level().m_skip_cutscene)
+      SoundManager::current()->stop_sounds();
+  }
+
+  session->get_current_level().m_is_in_cutscene = true;
+  session->get_current_level().m_skip_cutscene = false;
+}
+/**
+ * Ends a skippable cutscene.
+ */
+static void end_cutscene()
+{
+  auto session = GameSession::current();
+  if (session == nullptr)
+  {
+    log_info << "No game session." << std::endl;
+    return;
+  }
+
+  if (!session->get_current_level().m_is_in_cutscene)
+  {
+    log_warning << "end_cutscene(): no cutscene to end, resetting status anyways." << std::endl;
+  }
+
+  // Remove all sounds that started playing while skipping.
+  if (session->get_current_level().m_skip_cutscene)
+    SoundManager::current()->stop_sounds();
+
+  session->get_current_level().m_is_in_cutscene = false;
+  session->get_current_level().m_skip_cutscene = false;
+}
+/**
+ * Checks if a skippable cutscene is currently running.
+ */
+static bool check_cutscene()
+{
+  auto session = GameSession::current();
+  if (session == nullptr)
+  {
+    log_info << "No game session" << std::endl;
+    return false;
+  }
+
+  return session->get_current_level().m_is_in_cutscene;
+}
+
+/**
+ * Suspends the script execution for a specified number of seconds.
+ * @param float $seconds
+ */
+static SQInteger wait(HSQUIRRELVM vm, float seconds)
+{
+  ssq::VM* ssq_vm = ssq::VM::get(vm);
+  if (ssq_vm && !ssq_vm->isThread()) return 0;
+
+  auto session = GameSession::current();
+  if (session && session->get_current_level().m_skip_cutscene)
+  {
+    if (ssq_vm && ssq_vm->getForeignPtr())
+    {
+      auto squirrelenv = ssq_vm->getForeignPtr<SquirrelEnvironment>();
+      // Wait anyways, to prevent scripts like `while (true) {wait(0.1); ...}`.
+      return squirrelenv->wait_for_seconds(vm, 0);
```

Comment:
Is the second parameter supposed to be 0 instead of `seconds`?

Reply:
Yeah, I've not changed the arguments.